Mlem/App/Views/Pages/PostEditor/PostEditorView+LinkView.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  PostEditorView+LinkView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 29/08/2024.
+//
+
+import SwiftUI
+
+extension PostEditorView {
+    @ViewBuilder
+    var linkView: some View {
+        HStack {
+            Label(linkLabel, systemImage: Icons.websiteAddress)
+                .lineLimit(1)
+                .fontWeight(.semibold)
+                .foregroundStyle(palette.accent)
+                .padding(.leading, 8)
+            Spacer()
+            if link == .waiting {
+                Button {
+                    let url: URL?
+                    if let pastedUrl = UIPasteboard.general.url {
+                        url = pastedUrl
+                    } else if let pastedString = UIPasteboard.general.string, pastedString.starts(with: "http") {
+                        url = URL(string: pastedString, encodingInvalidCharacters: false)
+                    } else {
+                        return
+                    }
+                    if let url {
+                        link = .value(url)
+                    }
+                } label: {
+                    Label("Paste", systemImage: Icons.paste)
+                        .foregroundStyle(palette.selectedInteractionBarItem)
```

Comment:
For consistency with other suggestion about image buttons

```suggestion
                } label: {
                    Label("Paste", systemImage: Icons.paste)
                        .fontWeight(.semibold)
                        .foregroundStyle(palette.selectedInteractionBarItem)
```